### PR TITLE
Keehan

### PR DESCRIPTION
**Pull request by AlexKeehan**
Worked on tickets ETS-39, ETS-20, ETS-53. Should give us a mostly complete frontend view of how adding a training period is going to look

### CODE REVIEW

**Review by emikate — APPROVED** [review]
Looks good!

File: addTrainingPeriod.php [hunk 1]
```diff
@@ -25,17 +25,37 @@
...
+        $required = array(
+            "coursename",
+            "date",
+            "starttime",
+            "endtime",
+            "trainer",
+            "description",
+            "location",
+            "capacity"
+        );
```
Comment (ChrisC2020) [inline addTrainingPeriod.php:39]: Good naming scheme

File: addTrainingPeriod.php [hunk 2]
```diff
@@ -25,17 +25,37 @@
...
 
+        /*foreach ($args as $arg) {
+            echo "Arg ", $arg, "\n";
+        }
```
Comment (ChrisC2020) [inline addTrainingPeriod.php:43]: good use of echo for debugging, good that you commented it out when not needed.

File: addTrainingPeriod.php [hunk 3]
```diff
@@ -25,17 +25,37 @@
...
 
-        $option = isset($_POST['training-periods']) ? $_POST['training-periods'] : false;
+        foreach($required as $req) {
+            echo "Required ", $req, "\n";
+        }
+        */
```
Comment (ChrisC2020) [inline addTrainingPeriod.php:48]: good use of echo for debugging, good that it's commented out

File: addTrainingPeriod.php [hunk 4]
```diff
@@ -25,17 +25,37 @@
...
-        //    echo 'bad form data';
-        //    die();
-        //}
+        if (!wereRequiredFieldsSubmitted($args, $required)) {
+            echo 'bad form data';
+            die();
+        } else {
+            //Create new Course and insert into dbCourses
+        }
+
```
Comment (ChrisC2020) [inline addTrainingPeriod.php:58]: good error catching

File: addTrainingPeriod.php [hunk 5]
```diff
@@ -50,48 +70,45 @@
...
+                    <input type="text" id="trainer" name="trainer" required placeholder="Enter trainer name"> 
                     <label for="name">Description </label>
-                    <input type="text" id="description" name="' . $course . 'description" required placeholder="Enter description">
+                    <input type="text" id="description" name="description" required placeholder="Enter description">
                     <label for="name">Location </label>
-                    <input type="text" id="location" name="' . $course . 'location" required placeholder="Enter location">
+                    <input type="text" id="location" name="location" required placeholder="Enter location">
                     <label for="name">Volunteer Slots</label>
-                    <input type="text" id="capacity" name="' . $course . 'capacity" pattern="([1-9])|([01][0-9])|(20)" required placeholder="Enter a number">
+                    <input type="text" id="capacity" name="capacity" pattern="([1-9])|([01][0-9])|(20)" required placeholder="Enter a number">
```
Comment (ChrisC2020) [inline addTrainingPeriod.php:107]: Could use comments at the top of this to make it immediately apparent what this is

File: selectTrainingPeriod.php [hunk 6]
```diff
@@ -21,20 +21,28 @@
...
+        $difference = (int)$added_year - (int)$year;
+
         if (!preg_match("/^(\d{4})$/", $added_year, $year)) {
-            echo 'Incorrectly Formatted Year';
-            //header("Location: selectTrainingPeriod.php");
-            die();
+             $_SESSION['error'] = 'Incorrectly Formatted Year';
+        }
+        else if ($difference < 0) {
+            $_SESSION['error'] = 'Cannot Add Training Periods To Past Years';
```
Comment (ChrisC2020) [inline selectTrainingPeriod.php:35]: Nice error checking, but could've been handled better if you didn't use a text field for year input.

File: addTrainingPeriod.php (hunk 5 above)
Comment (AlexKeehan) [reply addTrainingPeriod.php:107]: Yep, will add those

File: selectTrainingPeriod.php (hunk 6 above)
Comment (AlexKeehan) [reply selectTrainingPeriod.php:35]: Yeah a year drop down menu is a better way of handling this

**Review by emikate — COMMENTED** [review]
Hey Alex, 

Your code looks great, there are no errors and it achieves the proper functionality. You did well with catching errors and handling them. The dynamic form generation based on user input is well done. I suggest adding more brief comments, specifically throughout the HTML section, so future developers can better understand the purpose/intentions of each section. Keep up the good work

Emily

**Pull request merged**